### PR TITLE
ref(app-store-connect-dialog): Applies Internal Design Feedback 

### PR DESCRIPTION
**Pull request by priscilawebdev**
Applies the following feedback:

- Displays the field in red with an error description if the App Store Connection validation does not pass.
- Adds an "accordion" component 
- In case of a new App Connect Store, the "App Connect Store credentials" panel will be expanded by default
- If the validation for 'App Connect Store credentials' or 'iTunes credentials' does not pass,  the corresponding panel will be expanded by default.

**Preview:**

image

image

image

ps: under the feature flag `organizations:app-store-connect`

### CODE REVIEW

File: static/app/views/settings/components/forms/richListField.tsx [hunk 1]
```diff
@@ -133,13 +134,13 @@ class RichList extends React.PureComponent<RichListProps> {
   };
 
   onRemoveItem = (item: ListItem, index: number) => {
-    if (!this.props.disabled) {
+    if (!this.props.disabled && this.props.onRemoveItem) {
       this.props.onRemoveItem(item, () => this.removeItem(index));
```
Comment (markstory) [inline static/app/views/settings/components/forms/richListField.tsx:138]:
```suggestion
    if (!this.props.disabled) {
      this.props?.onRemoveItem(item, () => this.removeItem(index));
```

File: static/app/views/settings/components/forms/richListField.tsx [hunk 2]
```diff
@@ -165,10 +166,19 @@ class RichList extends React.PureComponent<RichListProps> {
...
+
     return (
-      <Item disabled={disabled} key={index}>
-        {this.props.renderItem(item)}
-        {this.props.onEditItem && (
+      <Item disabled={!!disabled} error={!!error} key={index}>
+        {error && (
+          <ErrorIcon>
+            <Tooltip title={error}>
+              <IconWarning color="red300" />
```
Comment (markstory) [inline static/app/views/settings/components/forms/richListField.tsx:176]: Will this color be ok in darkmode?

File: static/app/views/settings/components/forms/richListField.tsx [hunk 3]
```diff
@@ -249,7 +259,38 @@ const ItemList = styled('ul')`
...
+
+  if (error) {
+    return `
+      color: ${theme.black};
+      border: 1px solid ${theme.red300};
+      &,
+      button,
+      button:hover,
+      button:focus {
+        background-color: ${theme.red100};
```
Comment (markstory) [inline static/app/views/settings/components/forms/richListField.tsx:281]: Should these pull from `theme.button.danger` or `theme.alert.error`?

File: static/app/views/settings/projectDebugFiles/externalSources/symbolSources.tsx [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+
+  if (
+    hasAppConnectStoreFeatureFlag &&
+    !hasSavedAppStoreConnect &&
+    !dropDownItems.find(dropDownItem => dropDownItem.value === 'appStoreConnect')
+  ) {
+    dropDownItems.push({
+      value: 'appStoreConnect',
+      label: t(DEBUG_SOURCE_TYPES.appStoreConnect),
+      searchKey: t('apple store connect'),
```
Comment (markstory) [inline static/app/views/settings/projectDebugFiles/externalSources/symbolSources.tsx:66]: Should this have 'itunes' or 'ios' in it?

File: static/app/views/settings/projectDebugFiles/externalSources/symbolSources.tsx [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+    if (symbolSourcesQuantity < symbolSources.length) {
+      return {
+        successMessage: t('Successfully removed custom repository'),
+        errorMessage: t('An error occured while removing custom repository'),
+      };
+    }
+
+    return {
+      successMessage: t('Successfully updated custom repository'),
+      errorMessage: t('An error occured while updating new custom repository'),
```
Comment (markstory) [inline static/app/views/settings/projectDebugFiles/externalSources/symbolSources.tsx:109]:
```suggestion
      errorMessage: t('An error occured while updating the custom repository'),
```

File: static/app/views/settings/projectDebugFiles/externalSources/symbolSources.tsx [hunk 6]
```diff
@@ -0,0 +1,212 @@
...
+      return {
+        successMessage: t('Successfully added custom repository'),
+        errorMessage: t('An error occured while adding new custom repository'),
+      };
+    }
+
+    if (symbolSourcesQuantity < symbolSources.length) {
+      return {
+        successMessage: t('Successfully removed custom repository'),
+        errorMessage: t('An error occured while removing custom repository'),
```
Comment (markstory) [inline static/app/views/settings/projectDebugFiles/externalSources/symbolSources.tsx:103]:
```suggestion
        errorMessage: t('An error occured while removing the custom repository'),
```

File: static/app/views/settings/projectDebugFiles/externalSources/symbolSources.tsx [hunk 7]
```diff
@@ -0,0 +1,212 @@
...
+      }
+      return symbolSource;
+    });
+  }
+
+  function getRequestMessages(symbolSourcesQuantity: number) {
+    if (symbolSourcesQuantity > symbolSources.length) {
+      return {
+        successMessage: t('Successfully added custom repository'),
+        errorMessage: t('An error occured while adding new custom repository'),
```
Comment (markstory) [inline static/app/views/settings/projectDebugFiles/externalSources/symbolSources.tsx:96]:
```suggestion
        errorMessage: t('An error occured while adding a new custom repository'),
```

File: static/app/views/settings/projectDebugFiles/externalSources/utils.tsx [hunk 8]
```diff
@@ -0,0 +1,10 @@
+import forEach from 'lodash/forEach';
+import set from 'lodash/set';
+
+export function unflattenKeys(obj: Record<string, string>): Record<string, any> {
```
Comment (markstory) [inline static/app/views/settings/projectDebugFiles/externalSources/utils.tsx:4]: Could call this `expandKeys`? :shrug:

File: static/app/views/settings/components/forms/richListField.tsx (hunk 2 above)
Comment (priscilawebdev) [reply static/app/views/settings/components/forms/richListField.tsx:176]: yes :)

File: static/app/views/settings/components/forms/richListField.tsx (hunk 3 above)
Comment (priscilawebdev) [reply static/app/views/settings/components/forms/richListField.tsx]: I've updated this component and removed this style

File: static/app/views/settings/projectDebugFiles/externalSources/buildInSymbolSources.tsx [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+  organization,
+  builtinSymbolSourceOptions,
+  builtinSymbolSources,
+  projectSlug,
+}: Props) {
+  function getRequestMessages(builtinSymbolSourcesQuantity: number) {
+    if (builtinSymbolSourcesQuantity > builtinSymbolSources.length) {
+      return {
+        successMessage: t('Successfully added built-in repository'),
+        errorMessage: t('An error occured while adding new built-in repository'),
```
Comment (markstory) [inline static/app/views/settings/projectDebugFiles/externalSources/buildInSymbolSources.tsx:28]:
```suggestion
        errorMessage: t('An error occurred while adding new built-in repository'),
```

File: static/app/views/settings/projectDebugFiles/externalSources/buildInSymbolSources.tsx [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+      };
+    }
+
+    return {
+      successMessage: t('Successfully removed built-in repository'),
+      errorMessage: t('An error occured while removing built-in repository'),
```
Comment (markstory) [inline static/app/views/settings/projectDebugFiles/externalSources/buildInSymbolSources.tsx:34]:
```suggestion
      errorMessage: t('An error occurred while removing built-in repository'),
```

File: static/app/views/settings/projectDebugFiles/externalSources/symbolSources.tsx [hunk 11]
```diff
@@ -0,0 +1,300 @@
...
+      value: symbolSourcesWithErrors,
+      errors: symbolSourcesErrors,
+    };
+  }
+
+  function getRequestMessages(symbolSourcesQuantity: number) {
+    if (symbolSourcesQuantity > symbolSources.length) {
+      return {
+        successMessage: t('Successfully added custom repository'),
+        errorMessage: t('An error occured while adding a new custom repository'),
```
Comment (markstory) [inline static/app/views/settings/projectDebugFiles/externalSources/symbolSources.tsx:148]:
```suggestion
        errorMessage: t('An error occurred while adding a new custom repository'),
```

File: static/app/views/settings/projectDebugFiles/externalSources/symbolSources.tsx [hunk 12]
```diff
@@ -0,0 +1,300 @@
...
+      };
+    }
+
+    if (symbolSourcesQuantity < symbolSources.length) {
+      return {
+        successMessage: t('Successfully removed custom repository'),
+        errorMessage: t('An error occured while removing the custom repository'),
```
Comment (markstory) [inline static/app/views/settings/projectDebugFiles/externalSources/symbolSources.tsx:155]:
```suggestion
        errorMessage: t('An error occurred while removing the custom repository'),
```

File: static/app/views/settings/projectDebugFiles/externalSources/symbolSources.tsx [hunk 13]
```diff
@@ -0,0 +1,300 @@
...
+      };
+    }
+
+    return {
+      successMessage: t('Successfully updated custom repository'),
+      errorMessage: t('An error occured while updating the custom repository'),
```
Comment (markstory) [inline static/app/views/settings/projectDebugFiles/externalSources/symbolSources.tsx:161]:
```suggestion
      errorMessage: t('An error occurred while updating the custom repository'),
```

**Review by markstory — APPROVED** [review]
Code looks good to me. Only a few small typo errors. I wasn't able to try the code out as I don't have an iTunes connect account.

**Comment by priscilawebdev** [comment]
Thanks for the feedback @markstory 🙏  since the JWT dependency bump was reverted https://github.com/getsentry/sentry/pull/26011 , the UI does not work at the moment. The backend team is working on it 😉

**Pull request merged**